File: examples/basic_window.cpp
#include "siwim.hpp"
using namespace siwim

int main() {
  Window win(1280, 720, "basic window");

  win.on_key_down = [&win](KeyEvent e) {
    if (e.key == Key.escape) {
      win.close();
    }
  }

  win.on_render = [&win](RenderEvent e) {
    Renderer r(win);
    r.clear(Color(0x202020));
    r.rectangle(100, 50, 300, 200, Color(0x40FF40));
  }

  win.run();

  return 0;
}

File: examples/basic_window.nim
import siwim

var win = newWindow(1280, 720, "basic window")

win.onKeyup = proc(e: KeyEvent) =
  if e.key == Key.escape:
    close win

win.onRender = proc(e: RenderEvent) =
  let r = win.renderer
  r.clear color(32, 32, 32)
  r.rectangle 100, 50, 300, 200, color(64, 255, 64)

run win
